### PR TITLE
Save Mach as Cfg after Selection

### PR DESCRIPTION
**Pull request by gretay-js**
- Add an option `select` to the existing compiler flag `-save-ir-after`. 
- If flag "-save-ir-after select" is provided, call `cfgize` after `Selection` to convert Mach to Cfg and then save it to a file.  

There is no file format to save Mach to a file (as an IR, not in textual form like we get from -dsel).  This is useful for debugging (cfg can be printed to .dot format that is easy to follow) and other tools that operate on CFGs can use it.

### CODE REVIEW

**Review by xclerc — APPROVED** [review]

File: backend/asmgen.ml [hunk 1]
```diff
@@ -71,14 +71,29 @@ let linear_unit_info =
...
-let cfg_unit_info =
+let new_cfg_unit_info () =
   { Cfg_format.unit_name = "";
     items = [];
     for_pack = None;
   }
 
+let cfg_unit_info = new_cfg_unit_info ()
+
+module Map = Map.Make(Compiler_pass)
```
[the remark this answers is not in the document]
Comment (gretay-js) [reply backend/asmgen.ml]: done

File: ocaml/utils/clflags.ml [hunk 2]
```diff
@@ -546,18 +546,20 @@ module Compiler_pass = struct
      - the manpages in man/ocaml{c,opt}.m
      - the manual manual/manual/cmds/unified-options.etex
   *)
-  type t = Parsing | Typing | Scheduling | Emit | Simplify_cfg
+  type t = Parsing | Typing | Scheduling | Emit | Simplify_cfg | Selection
```
[the remark this answers is not in the document]
Comment (gretay-js) [reply ocaml/utils/clflags.ml:549]: The type in the .mli is in the same order, can't see any other declarations. It was ordered "by rank" in some functions, I change it where possible.

File: ocaml/utils/clflags.ml [hunk 3]
```diff
@@ -582,12 +586,14 @@ module Compiler_pass = struct
...
     | Simplify_cfg -> true
+    | Selection -> true
     | Parsing | Typing -> false
 
   let enabled is_native t = not (is_native_only t) || is_native
   let can_save_ir_after = function
     | Scheduling -> true
     | Simplify_cfg -> true
+    | Selection -> true
     | _ -> false
```
[the remark this answers is not in the document]
Comment (gretay-js) [reply ocaml/utils/clflags.ml]: done

File: ocaml/utils/clflags.ml (hunk 2 above)
[the remark this answers is not in the document]
Comment (xclerc) [reply ocaml/utils/clflags.ml:549]: (By declaration, I meant type declaration and
value declaration like `passes`. It is still not the
same for `can_save_ir_after`, `is_native_only`, or
`rank`, but I don't know whether it matters or
not - feel free to resolve the conversation with
no change to the code.)

**Review by mshinwell — APPROVED** [review]

**Pull request merged**